### PR TITLE
Tanya/fixes

### PR DESCRIPTION
**Pull request by Tanya-atatakai**
## Proposed changes

Separate staking contract details and service info in order to fix start agent button

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: frontend/context/StakingContractDetailsProvider.tsx [hunk 1]
```diff
@@ -91,25 +94,48 @@ const useStakingContractDetailsByStakingProgram = ({
...
+          ),
         );
-      return serviceApi.getStakingContractDetailsByServiceIdStakingProgram(
-        serviceNftTokenId!,
-        stakingProgramId!,
-        evmHomeChainId,
-      );
+      }
+
+      return Promise.allSettled(promises).then((results) => {
```
Comment (Tanya-atatakai) [inline frontend/context/StakingContractDetailsProvider.tsx:117]: fixes this part was undefined 

image

File: frontend/service/agents/PredictTrader.ts [hunk 2]
```diff
@@ -141,70 +147,43 @@ export abstract class PredictTraderService extends StakedAgentService {
...
-      minStakingDeposit,
-    ] = multicallResponse;
-
-    const availableRewards = parseFloat(
-      ethers.utils.formatUnits(availableRewardsInBN, 18),
-    );
-    const serviceIds = getServiceIdsInBN.map(Number);
-
-    const maxNumServices = maxNumServicesInBN.toNumber();
+    const [serviceInfo, serviceStakingState] = multicallResponse;
```
[the remark this answers is not in the document]
Comment (Tanya-atatakai) [reply frontend/service/agents/PredictTrader.ts:170]: will keep as is here, according to the method used
```
stakingTokenProxy.getServiceInfo(serviceNftTokenId),
```

File: frontend/context/BalanceProvider.tsx [hunk 3]
```diff
@@ -105,9 +106,10 @@ export const BalanceProvider = ({ children }: PropsWithChildren) => {
 
   const totalStakedOlasBalance = useMemo(() => {
     return stakedBalances
-      .filter((walletBalance) => {
```
Comment (Tanya-atatakai) [inline frontend/context/BalanceProvider.tsx:108]: took me ~10 mins debugging and a lil crying to find out what's the problem here 😅

File: frontend/service/agents/PredictTrader.ts [hunk 4]
```diff
@@ -38,8 +42,11 @@ export abstract class PredictTraderService extends StakedAgentService {
 
     const provider = PROVIDERS[chainId].multicallProvider;
 
+    const mechContract =
+      MECHS[chainId][stakingProgramConfig.mechType!].contract;
+
     const contractCalls = [
-      // mechContract.getRequestsCount(agentMultisigAddress),
+      mechContract.getRequestsCount(agentMultisigAddress),
```
Comment (Tanya-atatakai) [inline frontend/service/agents/PredictTrader.ts:49]: idk where it's used, but we commented the call and forgot to comment the first item in the result, so everything was shifted 😄

File: frontend/context/BalanceProvider.tsx [hunk 5]
```diff
@@ -285,22 +293,23 @@ const getCrossChainWalletBalances = async (
...
 
-    if (!multisig || !token) {
+    if (!masterSafeAddress || !serviceNftTokenId) {
       return null;
     }
 
     const registryInfo = await StakedAgentService.getServiceRegistryInfo(
-      multisig,
-      token,
+      masterSafeAddress,
```
Comment (Tanya-atatakai) [inline frontend/context/BalanceProvider.tsx:311]: `getOperatorBalance` should use masterSafe address, not service multisig, that's why the bond part of staked balance was 0

File: frontend/context/BalanceProvider.tsx (hunk 3 above)
Comment (Tanya-atatakai) [reply frontend/context/BalanceProvider.tsx:108]: yes :DDD

**Review by truemiller — APPROVED** [review]

**Pull request merged**